infrastructure/smart-contracts/contracts/upgrades/esXai/esXai3.sol

```diff
@@ -303,4 +303,34 @@ contract esXai3 is ERC20Upgradeable, ERC20BurnableUpgradeable, AccessControlUpgr
         maxKeysNonKyc = newMax;
     }
 
+    /**
+    * @notice Retrieves a list of redemption requests for a specific user, starting from the highest index.
+    * @dev This function returns an array of `RedemptionRequestExt` structs for a given user,
+    *      starting from the highest index available and returning up to the maximum quantity requested.
+    * @param account The address of the user whose redemption requests are to be fetched.
+    * @param maxQty The maximum number of redemption requests to return.
+    * @param offset The offset from the highest index from which to start fetching the redemption requests.
+    * @return redemptions An array of `RedemptionRequestExt` structs containing the user's redemption requests in descending order by index.
+    * @return totalRedemptions The total number of redemption requests for the user.
+    */
+    function getRedemptionsByUser(address account, uint256 maxQty, uint256 offset) external view returns (RedemptionRequestExt[] memory redemptions, uint256 totalRedemptions) {
+        // Get the total number of redemption requests for the user.
+        totalRedemptions = _extRedemptionRequests[account].length;
+        
+        // Calculate the actual number of redemption requests to return, ensuring it does not exceed the array bounds.
+        uint256 qtyToReturn = maxQty > totalRedemptions ? totalRedemptions : maxQty;
+
+        // Create a dynamic array to hold the redemption requests to be returned.
+        redemptions = new RedemptionRequestExt[](qtyToReturn);
+
+        // Loop to fill the redemptions array starting from the highest index.
+        uint256 startIndex = totalRedemptions - offset - 1;
+        for (uint256 i = 0; i < qtyToReturn; i++) {
+            uint256 currentIndex = startIndex - i;
```

Comment:
Consider using `i--` going from the end of the array instead of using the `currentIndex`, in an EVM this could actually make a difference when trying to read more per request.

Reply:
Adjusted to remove additional calculations, thank you for the guidance!